flask_ansible.md

```diff
@@ -0,0 +1,182 @@
+
+You also need to have ansible installed on your local machine, that is, the machine you use to ssh into your droplet. You can find installation instructions for all platforms in the [official Ansible documentation](http://docs.ansible.com/ansible/intro_installation.html).
+
+## Why Ansible
+
+Using a provisioning framework has the advantage of letting us do a one-command deployment where we don't need to interact with the terminal. Ansible in particular is built around the concept of *indempotency* - the capability to run the same deployment script again and again and to have the certainty that the system will always reach the desired state safely, without unnecessarily redoing work that has already been done or throwing errors when a command is executed twice. 
+
+That means, for example, that commands such as "create a certain directory" are better expressed as "make sure that a certain directory is present in the system". Ansible works by letting you define "playbooks" which contain the commands that you need executed. The playbooks are written in [YAML](http://yaml.org/). With that in mind, we can get started writing our very own playbook to install Flask and deploy a sample application.
+
+## Building the Playbook
+
+First, we need to create a plain text file named flask.yml
+
+### Defining variables
+At the top of the file, we need to add the following lines:
+```
+- hosts: all
+
+  vars:
+    LOCAL_PROJECT_HOME: "./hello"
+    PROJECT_HOME: "~/my_test_home"
+    MAIN_PY: "hello.py" # main file to launch website
+    PORT: 5000 # port of the website
+```
+The first line signifies that we want to run this command on all the hosts that we are given. The deployment section will show how launch a playbook together with a host file to define the hosts for Ansible.
+
+  The vars section lists all the variables we're going to be using throught the playbook.
+
+### Flask Hello World
+
+We're going to be copying the local directory (LOCAL_PROJECT_HOME) to the PROJECT_HOME directory on the droplet. It contains the standard [hello world](http://flask.pocoo.org/docs/0.10/quickstart/) Flask project. The differences are that the hostname of the app is "0.0.0.0", so as to make it accessible externally and that it's possible to configure the port the website runs on via a command line argument. This is what hello.py looks like:
+
+```python
+import sys
+
+from flask import Flask
+
+app = Flask(__name__)
+
+@app.route('/')
+def hello_world():
+    return 'Hello World!'
+
+if __name__ == '__main__':
+    app.run(host="0.0.0.0", port=sys.argv[1])
+
+```
+
+After defining the top section, we merely need to list several task items for Ansible to execute. 
```

Comment:
those items should be listed under a task header
also maybe mention somewhere that yaml syntax is whitespace sensitive